### ISSUE TITLE
cannot disable checks right after "else:"

### ISSUE THREAD

**Issue by jwilk** [pylint-dev/pylint#9503]
### Bug description

```python
if True:
    pass
else:
    # pylint: disable=undefined-variable
    eggs
```

### Command used

```shell
pylint -E else.py
```

### Pylint output

```python
************* Module else
else.py:5:4: E0602: Undefined variable 'eggs' (undefined-variable)
```

### Expected behavior

No errors reported.

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.8.10 (default, Nov 22 2023, 10:22:35)
[GCC 9.4.0]
```

**Comment by DanielNoord** [comment]
Duplicate of https://github.com/pylint-dev/pylint/issues/872.

There is a fix but it never got reviewed.
